python/docs/source/getting_started/install.rst

```diff
@@ -165,16 +168,109 @@ To install PySpark from source, refer to |building_spark|_.
+
+Optional dependencies
+~~~~~~~~~~~~~~~~~~~~~
+
+PySpark has several optional dependencies that enhance its functionality for specific modules.
+These dependencies are only required for certain features and are not necessary for the basic functionality of PySpark.
+If these optional dependencies are not installed, PySpark will function correctly for basic operations but will raise an ``ImportError``
+when you try to use features that require these dependencies.
+
+Spark Connect
+^^^^^^^^^^^^^
+
+Installable with ``pip install "pyspark[connect]"``.
+
+========================== ================= ====================================================================
+Package                    Supported version Note
+========================== ================= ====================================================================
+`pandas`                   >=1.4.4           Required for Spark Connect.
+`pyarrow`                  >=10.0.0          Crucial for data serialization and network communication efficiency.
+`grpcio`                   >=1.62.0          Necessary for implementing RPC functionalities in Spark Connect.
+`grpcio-status`            >=1.62.0          Supports detailed status handling in network communications.
+`googleapis-common-protos` >=1.56.4          Provides essential Google API protocols for network interactions.
+========================== ================= ====================================================================
+
+Spark SQL
+^^^^^^^^^
+
+Installable with ``pip install "pyspark[sql]"``.
+
+========= ================= ===============================================================
+Package   Supported version Note
+========= ================= ===============================================================
+`pandas`  >=1.4.4           Used for pandas UDFs and ``DataFrame.toPandas``.
+`pyarrow` >=10.0.0          For faster data conversion/transfer between PySpark and Pandas.
+========= ================= ===============================================================
+
+
+Pandas API on Spark
+^^^^^^^^^^^^^^^^^^^
+
+Installable with ``pip install "pyspark[pandas_on_spark]"``.
+
+========= ================= ===============================================================
+Package   Supported version Note
+========= ================= ===============================================================
+`pandas`  >=1.4.4           Required for Pandas API on Spark.
+`pyarrow` >=10.0.0          For faster data conversion/transfer between PySpark and pandas.
+========= ================= ===============================================================
+
+Additional libraries that enhance functionality but are not included in the installation packages:
+
+- **mlflow**: Enhances machine learning lifecycle management. Used for ``pyspark.pandas.mlflow``.
+- **plotly**: Provide advanced plotting capabilities for visualization.
+- **matplotlib**: Provide plotting, but **plotly** is recommended. The default is **plotly**.
```

Comment:
Can you say `but **plotly** is recommended. The default is **plotly**.` at `**plotly**`?